js/modules/k6/http/response.go

```diff
@@ -109,8 +110,25 @@ func (res *HTTPResponse) Json(selector ...string) goja.Value {
 		}
 
 		if err := json.Unmarshal(body, &v); err != nil {
+			var msg string
+			switch t := err.(type) {
+			case *json.SyntaxError:
+				jsn := string(body[0:t.Offset])
+				jsn += color.RedString("<--(Invalid Character)")
+				jsn += string(body[t.Offset:])
+				msg = fmt.Sprintf("\n Invalid character at offset %v\n %s", t.Offset, jsn)
+			case *json.UnmarshalTypeError:
+				jsn := string(body[0:t.Offset])
+				jsn += "<--(Invalid Type)"
+				jsn += string(body[t.Offset+1:])
+				msg = fmt.Sprintf("\n Invalid value at offset %v\n %s", t.Offset, jsn)
+			default:
+				msg = err.Error()
+			}
+			err = errors.New(err.Error() + msg)
```

Comment:
Dumping the whole json seems like a bad idea. I would guess the majority of times this happen the problem is that actually what is given as json is not json at all. Probably error page of some kind and dumping it won't help much not to speak of how noisy it will be.

 I would prefer if we can maybe just say that there was an error at a given offset. Preferably line number and column but this might be harder.

Reply:
@MStoykov  Yeah that makes sense 👍 , I will work on it